proxy/http/Makefile.am

```diff
@@ -98,6 +99,26 @@ test_ForwardedConfig_LDADD = \
 	$(top_builddir)/src/tscore/libtscore.la \
 	$(top_builddir)/src/tscpp/util/libtscpputil.la
 
+test_error_page_selection_CPPFLAGS = $(AM_CPPFLAGS)\
```

Comment:
This should be more generic, e.g. something like "test_proxy_http". Generally we group unit tests per directory, as a compromise between completely separate (which would be a lot of compilation overhead) and all unit tests in one executable.

Reply:
OK, I'll add a main unit test file under this directory and group them up.

Reply:
Looks like I got some duplicate symbols from test_ForwardedConfig_mocks when I put them together.
```
duplicate symbol _ioBlockAllocator in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(IOBuffer.o)
duplicate symbol _ioBufAllocator in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(IOBuffer.o)
duplicate symbol _ioDataAllocator in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(IOBuffer.o)
duplicate symbol __ZN13IOBufferBlock4freeEv in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZN12IOBufferData4freeEv in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZN13IOBufferBlockC2Ev in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZN13IOBufferBlockC1Ev in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTV13IOBufferBlock in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTS13IOBufferBlock in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTI13IOBufferBlock in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTV12IOBufferData in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTS12IOBufferData in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTI12IOBufferData in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol _thread_freelist_low_watermark in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(ProxyAllocator.o)
duplicate symbol _thread_freelist_high_watermark in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(ProxyAllocator.o)
duplicate symbol __ZN6Thread15thread_data_keyE in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Thread.o)
ld: 16 duplicate symbols for architecture x86_64
```